### PR TITLE
Add version history for Bio-Formats 7.3.0

### PR DESCRIPTION
**Pull request by dgault**
Adding version history for the release of 7.3.0

For a full list of included PR's see https://github.com/ome/bioformats/milestone/100?closed=1

### CODE REVIEW

File: sphinx/about/whats-new.rst [hunk 1]
```diff
@@ -1,6 +1,55 @@
 Version history
 ===============
 
+7.3.0 (2024 April)
+------------------
+
+File format fixes and improvements:
+
+* CellSens VSI
+   - fixed and exception in CellSens 4.1 data by updating the handling of orphaned ets files (thanks to Edward Scanlon)
```
Comment (melissalinkert) [inline sphinx/about/whats-new.rst:10]:
```suggestion
   - fixed an exception in CellSens 4.1 data by updating the handling of orphaned ets files (thanks to Edward Scanlon)
```

File: sphinx/about/whats-new.rst [hunk 2]
```diff
@@ -1,6 +1,55 @@
...
+* added coverage of Java 21 builds as part of the CI process
+
+Component updates:
+
+* `ome-jai` was upgraded to 0.1.4
+* `ome-stubs` was upgraded to 6.0.2
+* `ome-mdbtools` was upgraded to 5.3.3
+* `ome-common` was upgraded to 6.0.22
+* `ome-poi` was upgraded to 5.3.9
+* `ome-metakit` was upgraded to 5.3.7
```
Comment (melissalinkert) [inline sphinx/about/whats-new.rst:43]: https://github.com/ome/bioformats/pull/4183 bumped ome-metakit to 5.3.6 (but 5.3.7 is latest).

File: sphinx/about/whats-new.rst [hunk 3]
```diff
@@ -1,6 +1,55 @@
...
+* added coverage of Apple M1 as part of the CI process
+* added coverage of Java 21 builds as part of the CI process
+
+Component updates:
+
+* `ome-jai` was upgraded to 0.1.4
+* `ome-stubs` was upgraded to 6.0.2
+* `ome-mdbtools` was upgraded to 5.3.3
+* `ome-common` was upgraded to 6.0.22
+* `ome-poi` was upgraded to 5.3.9
```
Comment (melissalinkert) [inline sphinx/about/whats-new.rst:42]: https://github.com/ome/bioformats/pull/4183 bumped ome-poi to 5.3.8 (but 5.3.9 is latest).

**Review by melissalinkert — CHANGES_REQUESTED** [review]
Noted a couple of mismatches between release notes and actual component versions. I suspect those are changes that need to happen in Bio-Formats itself.

**Comment by dgault** [comment]
Good spot @melissalinkert, in this case it is the docs which are correct. I have opened a further PR against bioformats to bump these 2 components to the latest version: https://github.com/ome/bioformats/pull/4185

**Review by sbesson — APPROVED** [review]

**Pull request merged**